File: app/src/main/java/de/hype/hypenotify/tools/pojav/DataResultReceiver.java
// File: src/main/java/com/example/otherapp/DataResultReceiver.java
package de.hype.hypenotify.tools.pojav;

import android.os.Bundle;
import android.os.Handler;
import android.os.ResultReceiver;

public class DataResultReceiver extends ResultReceiver {
    public interface Callback {
        void onDataReceived(String profiles, String accounts);
    }

    private Callback callback;

    public DataResultReceiver(Handler handler, Callback callback) {
        super(handler);
        this.callback = callback;
    }

    @Override
    protected void onReceiveResult(int resultCode, Bundle resultData) {
        String profiles = resultData.getString("profiles");
        String accounts = resultData.getString("accounts");
        if (callback != null) {
            callback.onDataReceived(profiles, accounts);
        }
    }
}

File: app/src/main/java/de/hype/hypenotify/app/tools/pojav/DataResultReceiver.kt
// File: src/main/java/com/example/otherapp/DataResultReceiver.java
package de.hype.hypenotify.app.tools.pojav

import android.os.Bundle
import android.os.Handler
import android.os.ResultReceiver

class DataResultReceiver(handler: Handler?, private val callback: Callback?) : ResultReceiver(handler) {
    interface Callback {
        fun onDataReceived(profiles: String?, accounts: String?)
    }

    override fun onReceiveResult(resultCode: Int, resultData: Bundle) {
        val profiles = resultData.getString("profiles")
        val accounts = resultData.getString("accounts")
        if (callback != null) {
            callback.onDataReceived(profiles, accounts)
        }
    }
}